task/deprecated-image-check/0.1/deprecated-image-check.yaml

```diff
@@ -33,62 +35,57 @@ spec:
+            echo "Response code: $http_code"
+            echo -n $http_code $IMAGE_REGISTRY $IMAGE_REPOSITORY \n>> $(results.PYXIS_HTTP_CODE.path)
+        done
 
     # Run the tests and save output
     - name: run-conftest
       image: quay.io/redhat-appstudio/hacbs-test:latest
       script: |
         #!/usr/bin/env sh
-        http_code=$(cat $(results.PYXIS_HTTP_CODE.path))
-
-        if [ "$http_code" == "200" ];
-        then
-          echo "Running conftest using $(params.POLICY_DIR) policy, $(params.POLICY_NAMESPACE) namespace"
-          /usr/bin/conftest test --no-fail $(workspaces.sanity-ws.path)/repository_data.json \
-          --policy $(params.POLICY_DIR) --namespace $(params.POLICY_NAMESPACE) \
-          --output=json 2> $(workspaces.sanity-ws.path)/stderr.txt | tee $(workspaces.sanity-ws.path)/deprecated_image_check_output.json
-          echo "Done!"
-          exit 0
-        elif [ "$http_code" == "404" ];
-        then
-          echo "Registry/image $(params.IMAGE_REGISTRY)/$(params.IMAGE_REPOSITORY) not found in Pyxis" > $(workspaces.sanity-ws.path)/stderr.txt
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        else
-          echo "Unexpected error (HTTP code $http_code) occured for registry/image $(params.IMAGE_REGISTRY)/$(params.IMAGE_REPOSITORY) during running conftest" > $(workspaces.sanity-ws.path)/stderr.txt
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        fi
-
-    # Format the test results and save them as the HACBS_TEST_OUTPUT Tekton result
-    - name: test-format-result
-      image: quay.io/redhat-appstudio/hacbs-test:latest
-      script: |
-        if [ ! -z $(cat $(workspaces.sanity-ws.path)/stderr.txt) ]
-        then
-          echo "The deprecated-image-check test meets the following error:"
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        fi
-
+        counter=0
+        cat $(results.PYXIS_HTTP_CODE.path)) | while read line;
+        do
+            IFS=:' ' read -r http_code IMAGE_REGISTRY IMAGE_REPOSITORY <<< $line; echo "[$http_code] [$IMAGE_REGISTRY] [$IMAGE_REPOSITORY]"
+            if [ "$http_code" == "200" ];
+            then
+              echo "Running conftest using $(params.POLICY_DIR) policy, $(params.POLICY_NAMESPACE) namespace"
+              /usr/bin/conftest test --no-fail $(workspaces.sanity-ws.path)/${IMAGE_REPOSITORY}/repository_data.json \
+              --policy $(params.POLICY_DIR) --namespace $(params.POLICY_NAMESPACE) \
+              --output=json 2>> $(workspaces.sanity-ws.path)/stderr.txt | tee $(workspaces.sanity-ws.path)/${IMAGE_REPOSITORY}/deprecated_image_check_output.json
+              echo "Done!"
+              exit 0
+            elif [ "$http_code" == "404" ];
+            then
+              echo "Registry/image $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY) not found in Pyxis" >> $(workspaces.sanity-ws.path)/stderr.txt
+              counter = $counter + 1
```

Comment:
I think that we should not count 404 results as failures since that only means that the repo is not in Pyxis and not that it's deprecated.

Reply:
done